Makefile

```diff
@@ -223,8 +225,53 @@ ifneq ($(GEM_INSTALL_OPTIONS),)
 	_GEM_INSTALL_OPTIONS = $(GEM_INSTALL_OPTIONS)
 endif
 
-# Should pay attention to warnings (some may be critical for crypto-enabled code (ex. signed-unsigned mismatch)
-CFLAGS += -Werror -Wno-switch
+define supported =
+$(shell if echo "int main(void){}" | $(if $(2),$(2),$(CC)) -x c -fsyntax-only -Werror $(1) - >/dev/null 2>&1; then echo "yes"; fi)
+endef
+
+# We are security-oriented so we use a pretty paranoid set of flags
+# by default. For starters, enable default set of warnings and treat
+# them as errors. Skip any unknown warning directives though.
+CFLAGS += -Wall -Wextra
```

Comment:
shall we wrap all these flags into some kind of "debug" flag? I'm not sure that regular users should build Themis with this level of paranoia, it's useful for us and our CIs only.

Reply:
Mmm... I'd say that we're pretty safe to use these flags even for user builds. We will see the warnings and should fix them so that any user build is warning-free. These are just warnings, they do not affect code generation, runtime performance or security. Furthermore, they affect only Themis build (and its wrappers), but they are not propagated onto their users. For example, these flags will not affect the code base using ThemisPP, they'll have to enable their own warnings if they wish to.

Reply:
After some discussion we have found a compromise. The warnings will be enabled for the users, but they won't be treated as errors. They are still reported to stderr during the build, but do not prevent successful compilation. However, our CI builds will still treat warnings as fatal errors.